### PR TITLE
Mention Web Parser

### PR DESCRIPTION
**Pull request by jakzaizzat**
### Summary of Changes

- Move the `ProcessText` logic to `shared`
- Create a new `GalleryProcessedText` component to handle all the text parser for caption/comment (+ support mentions)
- Render the mention in the `Post` and `Comment` in the web

Explanation on how it works is in the loom - https://www.loom.com/share/cc97994cc2f74a728c9bd31e4b1e281c

### Demo or Before/After Pics

https://github.com/gallery-so/gallery/assets/4480258/9e66216a-2ec1-4c0b-a835-3ba5e305e2b9

### Edge Cases

1. Render mentions on Post and comment
2. Render plain link on Post and comment
3. Render markdown link on Post and comment
4. Render all the 3 elements in one comment/post

### Testing Steps

1. You should able to see mention user/community in the feed and comment

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/src/components/Feed/Posts/PostHeader.tsx [hunk 1]
```diff
@@ -71,8 +76,9 @@ export default function PostHeader({ postRef, queryRef }: Props) {
       </HStack>
       <StyledBaseM>
         {post.caption && (
-          <Markdown
-            text={unescape(replaceUrlsWithMarkdownFormat(post.caption))}
+          <ProcessedText
+            text={unescape(post.caption)}
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/Posts/PostHeader.tsx:80]: nit: we no longer need to `unescape`, backend does it for us :)

File: apps/web/src/components/ProcessedText/elements/MentionComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import CommunityHoverCard from '~/components/HoverCard/CommunityHoverCard';
+import UserHoverCard from '~/components/HoverCard/UserHoverCard';
+import { MentionComponentFragment$key } from '~/generated/MentionComponentFragment.graphql';
+
+import { GalleryTextElementParserMentionsFragment$data } from '../../../../../../packages/shared/__generated__/relay/GalleryTextElementParserMentionsFragment.graphql';
```
Comment (Robinnnnn) [inline apps/web/src/components/ProcessedText/elements/MentionComponent.tsx:8]: let's make sure to do `'~/shared/...'`

**Review by Robinnnnn — APPROVED** [review]

File: apps/web/src/components/Feed/Posts/PostHeader.tsx (hunk 1 above)
Comment (jakzaizzat) [reply apps/web/src/components/Feed/Posts/PostHeader.tsx]: Removed it!

File: apps/web/src/components/ProcessedText/elements/MentionComponent.tsx (hunk 2 above)
Comment (jakzaizzat) [reply apps/web/src/components/ProcessedText/elements/MentionComponent.tsx]: Updated!

**Pull request merged**